changelogs/current.yaml

```diff
@@ -32,5 +32,8 @@ new_features:
 - area: tls
   change: |
     Added support for P-384 and P-521 curves for TLS server certificates.
+- area: stats ads
+  change: |
+    Fixed metric for ADS disconnection counters, extracting the tag from ``grpc.ads_cluster.*`` counter stats.
```

Comment:
Is this impacting prometheus in any way? (If so, please reflect this in the comment)

Reply:
@adisuissa it should. I was testing it now, though, and I don't see any GRPC disconnection stats. My `dyanamic_resources` config looks like this: 

```
dynamic_resources:
  ads_config:
    api_type: GRPC
    grpc_services:
    - envoy_grpc:
        cluster_name: xds_cluster
  cds_config:
    ads: {}
  lds_config:
    ads: {}
```

Anything I could be missing to enforce those stats (if they're implemented)?

Reply:
I was having a look into https://github.com/envoyproxy/envoy/blob/742a3b02e3b2a9dfb877a7e378607c6ed0c2aa53/source/extensions/config_subscription/grpc/grpc_stream.h#L111

It doesn't look like there's an implementation for that stat. Happy to implement it if you think it makes sense.

Reply:
I think that the counters you are referring to are the grpc ones, see [google_grpc](https://github.com/envoyproxy/envoy/blob/742a3b02e3b2a9dfb877a7e378607c6ed0c2aa53/source/common/grpc/google_async_client_impl.cc#L252), and envoy_grpc (which are the cluster stats).
I'm assuming that the tag extraction should be more generic, and not only for ADS, no?

Reply:
That makes sense. Though I can't get any stats from google_async_client_impl.cc with my configuration. Actually, I don't see any stats for GRPC at all when running the Envoy server.

Reply:
@adisuissa With the proper configuration now it all made sense to me. I'm quite new to Envoy so I totally missed your tip. I changed the prefix name to "google_grpc_client" so it's generic enough. Let me know what you think. And also, I've added a more detailed description of how Prometheus stats are impacted.